app/controllers/api/v1/import_controller.rb

```diff
@@ -0,0 +1,139 @@
+  def create
+    archivist = User.find_by_login(params[:archivist])
+    external_works = params[:works]
+    works_responses = []
+
+    # check for top-level errors (not an archivist, no works...)
+    status, messages = batch_errors(archivist, external_works)
+
+    if status == :ok
+      # Flag error and successes
+      @some_errors = @some_success = false
+
+      # Process the works, updating the flags
+      external_works.each do |external_work|
+        works_responses << import_work(archivist, external_work)
+      end
+
+      # To be added if required
+      # send_external_invites(@works)
+
+      # set final response code and message depending on the flags
+      status, messages = response_code(messages)
+    end
+    render status: status, json: { message: messages, works: works_responses }
+  end
+
+  private
+
+  # Set HTTP responses based on success and error flags
+  def response_code(messages)
+    if @some_success && @some_errors
+      status = :multi_status
+      messages << "At least one work was not imported. Please check the works array for further information."
+    elsif !@some_success && @some_errors
+      status = :unprocessable_entity
+      messages << "None of the works were imported. Please check the works array for further information."
+    else
+      status = :created
+      messages << "All works were successfully imported."
+    end
+    [status, messages]
+  end
+
+  # Use the story parser to import works from the chapter URLs,
+  # and set success or error flag depending on the outcome
+  # Returns a hash
+  def import_work(archivist, external_work)
+    work_status, work_messages = work_errors(external_work)
+    work_url = ""
+    original_url = []
+    if work_status == :ok
+      urls = external_work[:chapter_urls]
+      original_url = urls.first
+      storyparser = StoryParser.new
+      options = options(archivist, external_work)
+      begin
+        work = storyparser.download_and_parse_chapters_into_story(urls, options)
+        work.save
+        @some_success = true
+        work_status = :created
+        work_url = work_url(work)
+        work_messages << 'Successfully created work "' + work.title + '"'
+      rescue => exception
+        @some_errors = true
+        work_status = :unprocessable_entity
+        work_messages << exception.message
+      end
+    end
+
+    {
+      status: work_status,
+      url: work_url,
+      original_url: original_url,
+      messages: work_messages
+    }
+  end
+
+  # Top-level error handling: returns a 403 forbidden if a valid archivist isn't supplied and a 400
+  # if no works are supplied. If there is neither a valid archivist nor valid works, a 400 is returned
+  # ith both errors as a message
+  def batch_errors(archivist, external_works)
+    status = :bad_request
+    errors = []
+
+    unless archivist && archivist.is_archivist?
+      status = :forbidden
+      errors << "Only an Archive archivist can import works through the API."
```

Comment:
Unless I misunderstand, this seems like a potential source of confusion once more people are using -- it's that the user account you're importing to has to be an archivist, not the person using it who may not even be logged in, right? I'd suggest "You can only use the API to import works into an AO3 user account that has archivist privileges." maybe instead?